src/test/regress/sql/anonymous_columns.sql

```diff
@@ -7,9 +7,9 @@ CREATE TABLE t0 (a int PRIMARY KEY, b int, "?column?" text);
 SELECT create_distributed_table('t0', 'a');
 INSERT INTO t0 VALUES (1, 2, 'hello'), (2, 4, 'world');
 
-SELECT "?column?" FROM t0;
+SELECT "?column?" FROM t0 ORDER BY 1;
 
-WITH a AS (SELECT * FROM t0) SELECT "?column?" FROM a;
+WITH a AS (SELECT * FROM t0) SELECT "?column?" FROM a ORDER BY 1;
 WITH a AS (SELECT '' FROM t0) SELECT * FROM a;
```

Comment:
Currently we have a few of such tests `SELECT * FROM <table>` without ORDER BY and currently they don't return any data that could be different with order by. If we change what they return in the future, they might fail because we don't have ORDER BY. In general whenever there is a `SELECT * FROM...` it seems like it could be good to use ORDER BY to prevent such things from happening in the future. Do you think that is necessary in this test file? Is there a dowside in doing that?

Reply:
I don't think there is a downside to add the order by's.

For now I only solved the ordering for tests that could return in a different order, but let me add order by to the others as well.

Reply:
As discussed, while adding more ORDER BY clauses we ran into what turned out to be a longstanding bug in Citus where certain ORDER BY clauses causes the parser of postgres to trip up when parsing a deparsed query. https://github.com/citusdata/citus/issues/4030

Since the statements missing an ORDER BY will not change their output based on a race in the executor I think we can merge this fix for the flappiness now, before we find a solution for the ambiguous name error.

Reply:
Yeah that sounds good to me, thanks for creating the issue.